### PR TITLE
Update README with versioning, deprecation details

### PR DESCRIPTION
**Pull request by ysawa0**
This should resolve some of the discussion about versioning/deprecation in #257
- added a blurb about docker images and how they're tagged on Dockerhub
- fixed the "Deprecation Schedule" section to reflect sha rather than v2.0.0
- my markdown formatter auto-removed some extra whitespace

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,27 +48,27 @@ applications. Applications request a rate limit decision based on a domain and a
...
-3. `v2.0.0` deleted support for the legacy [ratelimit.proto](https://github.com/envoyproxy/ratelimit/blob/0ded92a2af8261d43096eba4132e45b99a3b8b14/proto/ratelimit/ratelimit.proto).
+3. `e91321b` [commit](https://github.com/envoyproxy/ratelimit/commit/e91321b10f1ad7691d0348e880bd75d0fca05758) deleted support for the legacy [ratelimit.proto](https://github.com/envoyproxy/ratelimit/blob/0ded92a2af8261d43096eba4132e45b99a3b8b14/proto/ratelimit/ratelimit.proto).
 The current version of ratelimit protocol is changed to [v3 rls.proto](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/service/ratelimit/v3/rls.proto)
 while [v2 rls.proto](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/service/ratelimit/v3/rls.proto) is still supported
 as a legacy protocol.
-4. `v3.0.0` deletes support for legacy [v2 rls.proto](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/service/ratelimit/v3/rls.proto)
+4. `TODO` deletes support for legacy [v2 rls.proto](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/service/ratelimit/v3/rls.proto)
```
Comment (mattklein123) [inline README.md:71]: IMO we can just go ahead and do this. It's been long enough. Do you want to do it?

Comment (ysawa0) [reply README.md:71]: That's fair. Sure, I can work on that in a separate PR and update README appropriately at that time as well.

Comment (mattklein123) [reply README.md:71]: Sure, sounds good, thanks.

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**